packages/core/src/libraries/quota.ts

```diff
@@ -156,9 +158,20 @@ export const createQuotaLibrary = (
       return;
     }
 
-    const { quota: fullQuota, usage: fullUsage } = await getTenantSubscriptionQuotaAndUsage(
-      cloudConnection
-    );
+    const {
+      planId,
+      quota: fullQuota,
+      usage: fullUsage,
+    } = await getTenantSubscriptionData(cloudConnection);
+
+    // Do not block Pro plan from adding add-on resources.
+    if (
+      isDevFeaturesEnabled &&
+      planId === ReservedPlanId.Pro &&
```

Comment:
how about enterprise plans?

Reply:
Aligned with implementations on other places, we only concern `pro` plan for now, will include enterprise plan later and also for other checks. In order to enable add-on self checkout for both Pro plan and enterprise plan, the condition should be `planId !== ReservedPlanId.Development && planId !== ReservedPlanId.Free`.